platform/darwin/src/MGLGeometry_Private.h

```diff
@@ -13,6 +13,18 @@
 CGRect MGLExtendRect(CGRect rect, CGPoint point);
 
 NS_INLINE mbgl::LatLng MGLLatLngFromLocationCoordinate2D(CLLocationCoordinate2D coordinate) {
+    if (std::isnan(coordinate.latitude)) {
+        [NSException raise:NSInvalidArgumentException format:@"latitude must not be NaN"];
+    }
+    if (std::isnan(coordinate.longitude)) {
+        [NSException raise:NSInvalidArgumentException format:@"longitude must not be NaN"];
+    }
+    if (std::abs(coordinate.latitude) > 90.0) {
```

Comment:
Can you be more specific about what "further up" means in this context, or what specific changes you'd recommend?